modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/internal/info/item/provider/test/ObjectEntryFolderInfoItemObjectProviderTest.java

```diff
@@ -0,0 +1,117 @@
+import com.liferay.info.item.ERCInfoItemIdentifier;
+import com.liferay.info.item.provider.InfoItemObjectProvider;
+import com.liferay.object.model.ObjectEntryFolder;
+import com.liferay.object.test.util.ObjectEntryFolderTestUtil;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.service.GroupLocalService;
+import com.liferay.portal.kernel.service.ServiceContextThreadLocal;
+import com.liferay.portal.kernel.test.rule.AggregateTestRule;
+import com.liferay.portal.kernel.test.rule.DeleteAfterTestRun;
+import com.liferay.portal.kernel.test.util.RandomTestUtil;
+import com.liferay.portal.kernel.test.util.ServiceContextTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Jhosseph Gonzalez
+ */
+@RunWith(Arquillian.class)
+public class ObjectEntryFolderInfoItemObjectProviderTest {
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Before
+	public void setUp() throws Exception {
+		_group = _groupLocalService.fetchGroup(TestPropsValues.getGroupId());
+
+		_objectEntryFolder = ObjectEntryFolderTestUtil.addObjectEntryFolder();
+
+		ServiceContextThreadLocal.pushServiceContext(
+			ServiceContextTestUtil.getServiceContext(
+				_group, TestPropsValues.getUserId()));
+	}
+
+	@After
+	public void tearDown() throws Exception {
+		ServiceContextThreadLocal.popServiceContext();
+	}
+
+	@Test
+	public void testGetInfoItem() throws Exception {
+		Assert.assertEquals(
+			_objectEntryFolder,
+			_objectEntryFolderInfoItemObjectProvider.getInfoItem(
+				_group.getGroupId(),
+				new ClassPKInfoItemIdentifier(
+					_objectEntryFolder.getObjectEntryFolderId())));
+		Assert.assertEquals(
+			_objectEntryFolder,
+			_objectEntryFolderInfoItemObjectProvider.getInfoItem(
+				_group.getGroupId(),
+				new ERCInfoItemIdentifier(
+					_objectEntryFolder.getExternalReferenceCode())));
+		Assert.assertEquals(
+			_objectEntryFolder,
+			_objectEntryFolderInfoItemObjectProvider.getInfoItem(
+				RandomTestUtil.randomLong(),
+				new ERCInfoItemIdentifier(
+					_objectEntryFolder.getExternalReferenceCode(),
+					_group.getExternalReferenceCode())));
+
+		try {
+			_objectEntryFolderInfoItemObjectProvider.getInfoItem(
+				RandomTestUtil.randomLong(),
+				new ERCInfoItemIdentifier(
+					_objectEntryFolder.getExternalReferenceCode()));
+
+			Assert.fail();
+		}
+		catch (NoSuchInfoItemException noSuchInfoItemException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(noSuchInfoItemException);
+			}
+		}
```

Comment:
can we use that `AssertUtils.assertFailure()` here? there is an error message, so we should use it too.